dashboard/utils.py

```diff
@@ -55,12 +55,12 @@ def __init__(self, user, program, enrollments, current_grades, certificates):
         self.financial_aid_available = program.financial_aid_availability
 
         with transaction.atomic():
-            self.course_ids = list(CourseRun.objects.filter(course__program=self.program).exclude(
+            self.course_ids = set(CourseRun.objects.filter(course__program=self.program).exclude(
```

Comment:
The changes in this file are unrelated to the rest of the pull request, but were necessary to make the tests pass. This query does not have an `ORDER BY` clause, so the order of the results is undefined. Some of our tests were comparing this `course_ids` property to a list, where the ordering is important: these tests were only passing coincidentally. My other changes apparently made the database decide to return query results in a different order, which is totally legal, but caused those tests to fail.

I checked with @giocalitri, who confirmed that this variable is only used to check whether a course ID is present in this list or not, so a set is a better data type to use, anyway. Sets can be compared without regard to ordering, so it doesn't matter what order the database decides to return the results.